examples/plot_sleep_staging.py

```diff
@@ -0,0 +1,328 @@
+#
+# .. note::
+#    To load your own datasets either via MNE or from
+#    preprocessed X/y numpy arrays, see the `MNE Dataset
+#    Tutorial <./plot_mne_dataset_example.html>`__ and the `Numpy Dataset
+#    Tutorial <./plot_custom_dataset_example.html>`__.
+#
+
+from braindecode.datasets.sleep_physionet import SleepPhysionet
+
+dataset = SleepPhysionet(
+    subject_ids=[0, 1], recording_ids=[1], crop_wake_mins=30)
+
+
+######################################################################
+# Preprocessing
+# ~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# Next, we preprocess the raw data. We apply convert the data to microvolts and
+# apply a lowpass filter. We omit the downsampling step of [1]_ as the Sleep
+# Physionet data is already sampled at a lower 100 Hz.
+#
+
+from braindecode.datautil.preprocess import (
+    MNEPreproc, NumpyPreproc, preprocess)
+
+high_cut_hz = 30
+
+preprocessors = [
+    # convert from volt to microvolt, directly modifying the numpy array
+    NumpyPreproc(fn=lambda x: x * 1e6),
+    # bandpass filter
+    MNEPreproc(fn='filter', l_freq=None, h_freq=high_cut_hz),
+]
+
+# Transform the data
+preprocess(dataset, preprocessors)
+
+
+######################################################################
+# Extract windows
+# ~~~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# We extract 30-s windows to be used in the classification task.
+
+from braindecode.datautil.windowers import create_windows_from_events
+
+
+mapping = {  # We merge stages 3 and 4 following AASM standards.
+    'Sleep stage W': 0,
+    'Sleep stage 1': 1,
+    'Sleep stage 2': 2,
+    'Sleep stage 3': 3,
+    'Sleep stage 4': 3,
+    'Sleep stage R': 4
+}
+
+window_size_s = 30
+sfreq = 100
+window_size_samples = window_size_s * sfreq
+
+windows_dataset = create_windows_from_events(
+    dataset, trial_start_offset_samples=0, trial_stop_offset_samples=0,
+    window_size_samples=window_size_samples,
+    window_stride_samples=window_size_samples, preload=True, mapping=mapping)
+
+
+######################################################################
+# Window preprocessing
+# ~~~~~~~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# We also preprocess the windows by applying channel-wise z-score normalization
+# in each window.
+#
+
+from braindecode.datautil.preprocess import zscore
+
+preprocess(windows_dataset, [MNEPreproc(fn=zscore)])
+
+
+######################################################################
+# Split dataset into train and valid
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# We can easily split the dataset using additional info stored in the
+# `description` attribute of :class:`braindecode.datasets.BaseDataset`,
+# in this case ``session`` column. Here, we split the examples per subject.
```

Comment:
```suggestion
# in this case using the ``subject`` column. Here, we split the examples per subject.
```

there is no session used here...

Reply:
arrh thanks for spotting that @robintibor